File: src/dataset/cpp/collatz-conjecture_exercism-jpcenteno/openAI/ChatGPT4_collatz_conjecture_v4.cpp
#include "collatz_conjecture.h"

#include <stdexcept>

namespace collatz_conjecture {

    std::int64_t steps(std::int64_t n) {
        if (n < 1) {
            throw std::domain_error("n must be positive");
        }

        std::int64_t i = 0;
        while (n != 1) {
            n = (n % 2 == 0) ? (n / 2) : (3 * n + 1);
            ++i;
        }

        return i;
    }

}  // namespace collatz_conjecture

File: src/dataset/typescript/collatz-conjecture_exercism-samajammin/openAI/ChatGPT4_collatz-conjecture_v4.ts
class CollatzConjecture {
  static steps(int: number): number {
    if (int <= 0) throw new Error('Only positive numbers are allowed');
    let steps = 0;
    while (int !== 1) {
      steps++;
      int = (int & 1) === 0 ? int >> 1 : int * 3 + 1;
    }
    return steps;
  }
}

export default CollatzConjecture;